lib/mayaUsd/render/MaterialXGenOgsXml/Nodes/SurfaceNodeMaya.cpp

```diff
@@ -43,6 +43,47 @@ void SurfaceNodeMaya::emitLightLoop(
     const ShaderNode*  bsdf = bsdfInput->getConnectedSibling();
 
     shadergen.emitComment("Light loop", stage);
+
+#if MAYA_LIGHTAPI_VERSION_2 == 3
+    // MayaUSD issue #2121: Flat lighting not used in Maya 2022.3
+    //
+    // MaterialX has no concept of flat lighting or ambient lights, so we must find something that
+    // looks like a diffuse node and see what color it has.
```

Comment:
This is an ugly hack, but there are no easy "give me the diffuse color" in MaterialX for a generic shader graph. Will give incorrect results for layered shaders where more than one diffuse node will be found.